File: backend/app/controllers/questions.py
from sqlalchemy.orm import Session
from app.models.questions import Question
from app.schemas.questions import QuestionCreate, QuestionUpdate





# function for getting all questions for a quiz
def get_questions_for_quiz(db: Session, quiz_id: int):
    return db.query(Question).filter(Question.quiz_id == quiz_id).all()

# function for getting a question by id
def get_question_by_id(db: Session, question_id: int):
    return db.query(Question).filter(Question.question_id == question_id).first()

# function for creating a new question
def create_question(db: Session, question: QuestionCreate):
    db_question = Question(
        question_text=question.question_text,
        quiz_id=question.quiz_id
    )
    db.add(db_question)
    db.commit()
    db.refresh(db_question)
    return db_question

# function for updating an existing question by id
def update_question(db: Session, question_id: int, question: QuestionUpdate):
    db_question = db.query(Question).filter(Question.question_id == question_id).first()
    db_question.question_text = question.question_text
    db_question.quiz_id = question.quiz_id
    db.commit()
    db.refresh(db_question)
    return db_question

# function for deleting an existing question by id
def delete_question(db: Session, question_id: int):
    question = db.query(Question).filter(Question.question_id == question_id).first()
    db.delete(question)
    db.commit()
    return question

File: front/src/store/questions.js
import instance from "@/middlewares";


export default {
  name: "questions",
  state: () => ({
    data: null,
    one_question: null,

  }),
  mutations: {
    setData(state, data) {
      state.data = data;
    },
    setOneQuestion(state, data) {
      state.one_question = data;
    },

  },
  actions: {
    // class Question(Base):
    // __tablename__ = "questions"

    // question_id = Column(BigInteger, primary_key=True, index=True)
    // question_text = Column(Text, nullable=False)
    // quiz_id = Column(BigInteger, ForeignKey("quizzes.quiz_id"), nullable=False)
    
    // quiz = relationship("Quiz", back_populates="question")
    // answer = relationship("Answer", back_populates="question", cascade="all, delete")

    // question crud
    // get questions
    async getQuestions({ commit }, quiz_id) {
      try {
        const response = await instance.get(`/api/questions/${quiz_id}`);
        if (response) return commit("setData", response.data);
      } catch (error) {
        console.log(error);
      }
    },
    // get question by id
    async getQuestionById({ commit }, id) {
      try {
        const response = await instance.get(`/api/questions/search/${id}`);
        if (response) return commit("setOneQuestion", response.data);
      } catch (error) {
        console.log(error);
      }
    },

    // create question
    async createQuestion({}, input) {
      try {
        const { question_text, quiz_id } = input;
        const response = await instance.post("/api/questions", { question_text, quiz_id });
        if (response.ok) return console.log("ok");
      } catch (error) {
        console.log(error);
      }
    },

    // update question
    async updateQuestion({}, input) {
      try {
        const { id, question_text, quiz_id } = input;
        const response = await instance.put(`/api/questions/${id}`, { question_text, quiz_id });
        if (response.ok) return console.log("ok");
      } catch (error) {
        console.log(error);
      }
    },

    // delete question
    async deleteQuestion({}, id) {
      try {
        const response = await instance.delete(`/api/questions/${id}`);
        if (response.ok) return console.log("ok");
      } catch (error) {
        console.log(error);
      }
    },

},

namespaced: true,
};
